lib/validate.js

```diff
@@ -94,8 +94,8 @@ exports.usingChecksum = (deviceFileDescriptor, options = {}) => {
     imageSize: options.imageSize,
     progress: options.progress,
     chunkSize: options.chunkSize
-  }).then((deviceChecksum) => {
-    if (options.imageChecksum !== deviceChecksum) {
+  }).then((results) => {
+    if (options.imageChecksum !== results.crc32) {
```

Comment:
This part still needs some figuring out on how we want to support checking one or multiple image checksums, especially as `.imageChecksum` won't tell us it's type (we could theoretically infer it from it's byte count, but that's not safe).